### PR TITLE
Create CF Mixing Water.mdx

### PR DESCRIPTION
**Pull request by jessechoe10**

### CODE REVIEW

File: solutions/CF Mixing Water.mdx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+			cout << 2 * maximumCups + 3 << endl;
+		}
+
+	}
+
+}
+
+int main(){
+	int t; cin >> t;
+	for(int i=1; i<=t; i++){
```
Comment (maggie-j-liu) [inline solutions/CF Mixing Water.mdx:124]:
```suggestion
	for(int i = 1; i <= t; i++){
```

File: solutions/CF Mixing Water.mdx [hunk 2]
```diff
@@ -0,0 +1,132 @@
+---
+id: cf-1359C
+source: CF
+title: Mixing Water
+author: Jesse Choe
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/78116)
+
+**Note: the $\mathcal{O}(1)$ solution for this problem is provided in the
```
Comment (maggie-j-liu) [inline solutions/CF Mixing Water.mdx:10]: maybe put this in an info block?

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
just checked code style, didn't check actual content lol

File: solutions/CF Mixing Water.mdx (hunk 2 above)
Comment (jessechoe10) [reply solutions/CF Mixing Water.mdx:10]: How do I do that?

Comment (maggie-j-liu) [reply solutions/CF Mixing Water.mdx:10]: https://usaco.guide/general/working-mdx/#info-block

File: solutions/CF Mixing Water.mdx [hunk 3]
```diff
@@ -36,41 +38,45 @@ using namespace std;
...
+		
 			maximumCups = mid;
 			left = mid + 1;
+			
 		} else {
+		
 			right = mid - 1;
+			
 		}
+
```
Comment (SansPapyrus683) [inline solutions/CF Mixing Water.mdx:79]: are all these extra lines intended lmao

Comment (maggie-j-liu) [reply solutions/CF Mixing Water.mdx:79]: I think you misinterpreted; add white space between parentheses and curly brackets, not the lines lol

File: solutions/CF Mixing Water.mdx [hunk 4]
```diff
@@ -0,0 +1,138 @@
+---
+id: cf-1359C
+source: CF
+title: Mixing Water
+author: Jesse Choe
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/78116)
```
Comment (maggie-j-liu) [inline solutions/CF Mixing Water.mdx:8]: i think you can delete this, it's probably not necessary to link it in two places

File: solutions/CF Mixing Water.mdx [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+			cout << 2 * maximumCups + 1 << endl;
+		} else {
+			cout << 2 * maximumCups + 3 << endl;
+		}
+	}
+}
+
+int main(){
+	int t; cin >> t;
+	for(int i = 1; i <= t; i++){
```
Comment (maggie-j-liu) [inline solutions/CF Mixing Water.mdx:110]:
```suggestion
int main() {
	int t; cin >> t;
	for(int i = 1; i <= t; i++) {
```

File: solutions/CF Mixing Water.mdx [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+void solve() {
+	/* 
+	   * maximumCups represents the the maximum number of cups such
+	   * that the barrel temperature is greater than desired_temp.
+	*/
+	cin >> hot_water >> cold_water >> desired_temp;
+	// Binary search on the answer for maximumCups.
+	int left = 0, right = 1e6 + 1, maximumCups = 0;
+
+	while(left <= right) {
```
Comment (SansPapyrus683) [inline solutions/CF Mixing Water.mdx:60]: maybe also spaces between the control statements and the parens? (`while (left <= right)`)?

Comment (frodakcin) [reply solutions/CF Mixing Water.mdx:60]: as long as it's consistent it's fine.

File: solutions/CF Mixing Water.mdx [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+---
+
+<Info title="Constant Time Solution">
+**Note: the $\mathcal{O}(1)$ solution for this problem is provided in the
+[official editorial](https://codeforces.com/blog/entry/78116).**
+</Info>
+
+Consider the following observations:
+
+- If the number of cups of hot water is equal to the number of cups of cold water in the barrel, then the minimum number of cups required to pour into the barrel is exactly $2$.
```
Comment (frodakcin) [inline solutions/CF Mixing Water.mdx:15]:
```suggestion
- Case 1: pouring an equal number of cups of hot and cold water is equivalent to pouring one cup of each. Otherwise,
```

File: solutions/CF Mixing Water.mdx [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+- The number of cups of hot water must be greater than the number of cups of cold water if the previous condition isn't satisfied. Thus, there will be $c + 1$ cups of hot water and $c$ cups of cold water.
```
Comment (frodakcin) [inline solutions/CF Mixing Water.mdx:16]:
```suggestion
- Case 2: there must be exactly one more cup of hot water poured than cold water poured. Thus, there will be $c + 1$ cups of hot water and $c$ cups of cold water for some $c \geq 0$.
```

File: solutions/CF Mixing Water.mdx [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+- The two barrels with the closest temperature to $t$ will be the barrels with temperatures slightly more and slightly less than $t$. The two barrels will be $a$ and $b$, respectively.
```
Comment (frodakcin) [inline solutions/CF Mixing Water.mdx:17]: Can delete this and say smth like since temperature func of c is decreasing we can binsearch to approximate it.

Comment (frodakcin) [reply solutions/CF Mixing Water.mdx:17]: ig you'll have to change the stuff below to flow better

**Review by frodakcin — CHANGES_REQUESTED** [review]

File: solutions/CF Mixing Water.mdx (hunk 9 above)
Comment (jessechoe10) [reply solutions/CF Mixing Water.mdx:17]: > Can delete this and say smth like since temperature func of c is decreasing we can binsearch to approximate it.

What?

Comment (frodakcin) [reply solutions/CF Mixing Water.mdx:17]: the temperature is l+(h-l)(c+1)/(2c+1) when c is odd. THis is a decreasing function (aka monotonic); since it's monotonic we can binsearch to approximate the best value for $c$, and test the odd integers aroudn it

File: solutions/CF Mixing Water.mdx [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+<Info title="Constant Time Solution">
+**Note: the $\mathcal{O}(1)$ solution for this problem is provided in the
+[official editorial](https://codeforces.com/blog/entry/78116).**
+</Info>
+
+Consider the following observations:
+
+- Case 1: pouring an equal number of cups of hot and cold water is equivalent to pouring one cup of each. Otherwise,
+- Case 2: there must be exactly one more cup of hot water poured than cold water poured. Thus, there will be $c + 1$ cups of hot water and $c$ cups of cold water for some $c \geq 0$.
+- The temperature is l+(h-l)(c+1)/(2c+1) when c is odd. This can be proven to be a monotonically decreasing function. Since it's monotonic, we can binary search to approximate the best value for $c$, and test the odd integers aroudn it
```
Comment (frodakcin) [inline solutions/CF Mixing Water.mdx:17]: Don't just directly copy it :/

I meant to just phrase it somewhere along these lines.

**Review by frodakcin — CHANGES_REQUESTED** [review]
Also why is build failing?

**Comment by jessechoe10** [comment]
> Also why is build failing?

i have no idea...

File: solutions/CF Mixing Water.mdx (hunk 10 above)
Comment (jessechoe10) [reply solutions/CF Mixing Water.mdx:17]: Should I change it?

File: solutions/CF Mixing Water.mdx [hunk 11]
```diff
@@ -0,0 +1,116 @@
+---
+id: cf-1359C
```
Comment (maggie-j-liu) [inline solutions/CF Mixing Water.mdx:2]: change the filename to match the id

File: solutions/CF Mixing Water.mdx [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+
+using namespace std;
+
+double hot_water, cold_water, desired_temp;
+
+/*
+	* This function checks whether pouring cupsPoured amount of cups achieves an average
+	* barrel temperature greater than or equal to desired_temp. Technically, this represents
+	* 2 * cupsPoured + 1 cups
+*/
```
Comment (maggie-j-liu) [inline solutions/CF Mixing Water.mdx:40]:
```suggestion
/*
   * This function checks whether pouring cupsPoured amount of cups achieves an average
   * barrel temperature greater than or equal to desired_temp. Technically, this represents
   * 2 * cupsPoured + 1 cups
*/
```
to match the comments below

File: solutions/CF Mixing Water.mdx [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+		   * greater than desired_temp. barrel2 represent the barrel with an average
+		   * temperature slightly less than desired_temp. These two barrels will be the
+		   * closest to desired_temp.
+		*/
+		double totalBarrel1 = (hot_water * (maximumCups + 1) + cold_water * (maximumCups));
+		double totalBarrel2 = (hot_water * (maximumCups + 2) + cold_water * (maximumCups + 1));
+		double barrel1Cups = (double)(2 * maximumCups + 1.0);
+		double barrel2Cups = (double)(2 * maximumCups + 3.0);
+		double avgBarrel1 = (totalBarrel1) / barrel1Cups;
+		double avgBarrel2 = (totalBarrel2) / barrel2Cups;
```
Comment (maggie-j-liu) [inline solutions/CF Mixing Water.mdx:84]:
```suggestion
		double totalBarrel1 = (hot_water * (maximumCups + 1) + cold_water * maximumCups);
		double totalBarrel2 = (hot_water * (maximumCups + 2) + cold_water * (maximumCups + 1));
		double barrel1Cups = (double)(2 * maximumCups + 1.0);
		double barrel2Cups = (double)(2 * maximumCups + 3.0);
		double avgBarrel1 = totalBarrel1 / barrel1Cups;
		double avgBarrel2 = totalBarrel2 / barrel2Cups;
```

File: solutions/CF Mixing Water.mdx [hunk 14]
```diff
@@ -0,0 +1,116 @@
...
+		double hot_cold_avg = (hot_water + cold_water) / 2.0;
+
+		double minAbsDifference = min(abs(desired_temp - avgBarrel1),
+		min(abs(desired_temp - avgBarrel2),
+		abs(desired_temp - hot_cold_avg)));
```
Comment (maggie-j-liu) [inline solutions/CF Mixing Water.mdx:89]:
```suggestion
		double minAbsDifference = min({abs(desired_temp - avgBarrel1),
										abs(desired_temp - avgBarrel2),
										abs(desired_temp - hot_cold_avg)});
```
not sure how to indent so it looks nice

**Review by maggie-j-liu — CHANGES_REQUESTED** [review]

File: content/3_Silver/Conclusion.problems.json [hunk 15]
```diff
@@ -46,6 +46,11 @@
       "url": "http://usaco.org/index.php?page=viewproblem2&cpid=225",
       "source": "Old Bronze",
       "difficulty": "Hard",
+      "uniqueId": "cf-371C",
+      "name": "Hamburgers",
+      "url": "https://codeforces.com/problemset/problem/371/C",
+      "source": "CF",
+      "difficulty": "Easy",
```
Comment (maggie-j-liu) [inline content/3_Silver/Conclusion.problems.json:53]: what's happening here?

File: content/3_Silver/Conclusion.problems.json [hunk 16]
```diff
@@ -31,10 +30,11 @@
       "uniqueId": "cf-1174D",
       "name": "Ehab and the Expected XOR Problem",
       "url": "https://codeforces.com/contest/1174/problem/D",
+
       "source": "CF",
-      "difficulty": "Normal",
+      "difficulty": "Easy",
       "isStarred": false,
-      "tags": ["Prefix Sums"],
+      "tags": ["Binary Search"],
```
Comment (maggie-j-liu) [inline content/3_Silver/Conclusion.problems.json:37]: why these changes?

Comment (jessechoe10) [reply content/3_Silver/Conclusion.problems.json:37]: Idk I just copied and pasted from CPI/usaco-guide. You can fix it yourself.

Comment (maggie-j-liu) [reply content/3_Silver/Conclusion.problems.json]: ok i think i fixed it

File: solutions/CF Mixing Water.mdx (hunk 10 above)
Comment (frodakcin) [reply solutions/CF Mixing Water.mdx]: Yes... its very hard to read atm

**Comment by jessechoe10** [comment]
Ok @frodakcin I changed it. Can you review?

File: solutions/cf-1359C.mdx [hunk 17]
```diff
@@ -0,0 +1,114 @@
...
+#include <bits/stdc++.h>
+
+using namespace std;
+
+double hot_water, cold_water, desired_temp;
+
+/*
+   * This function checks whether pouring cupsPoured amount of cups achieves an average
+   * barrel temperature greater than or equal to desired_temp. Technically, this represents
+   * 2 * cupsPoured + 1 cups
```
Comment (SansPapyrus683) [inline solutions/cf-1359C.mdx:37]: multiline comments are kinda weird

File: solutions/cf-1359C.mdx [hunk 18]
```diff
@@ -0,0 +1,114 @@
...
+		double totalBarrel2 = (hot_water * (maximumCups + 2) + cold_water * (maximumCups + 1));
+		double barrel1Cups = (double)(2 * maximumCups + 1.0);
+		double barrel2Cups = (double)(2 * maximumCups + 3.0);
+		double avgBarrel1 = totalBarrel1 / barrel1Cups;
+		double avgBarrel2 = totalBarrel2 / barrel2Cups;
+		double hot_cold_avg = (hot_water + cold_water) / 2.0;
+
+		double minAbsDifference = min({abs(desired_temp - avgBarrel1),
+										abs(desired_temp - avgBarrel2),
+										abs(desired_temp - hot_cold_avg)});
```
Comment (SansPapyrus683) [inline solutions/cf-1359C.mdx:87]: i don't think this formatting is correct

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
just some formatting issues and that's it

File: solutions/cf-1359C.mdx [hunk 19]
```diff
@@ -0,0 +1,114 @@
...
+
+<Info title="Constant Time Solution">
+**Note: the $\mathcal{O}(1)$ solution for this problem is provided in the
+[official editorial](https://codeforces.com/blog/entry/78116).**
+</Info>
+
+Consider the following observations:
+
+- Case 1: pouring an equal number of cups of hot and cold water is equivalent to pouring one cup of each. Otherwise,
+- Case 2: there must be exactly one more cup of hot water poured than cold water poured. Thus, there will be $c_i + 1$ cups of hot water and $c_i$ cups of cold water for some $c \geq 0$.
```
Comment (maggie-j-liu) [inline solutions/cf-1359C.mdx:16]: these are not really observations or cases (i think the first is an observation and the second is a case)...maybe
```suggestion
- Case 1: The number of cups of hot water poured is equal to the number of cups of cold water poured. Pouring an equal number of cups of hot and cold water is equivalent to pouring one cup of each. 
- Case 2: Otherwise, there must be exactly one more cup of hot water poured than cold water poured. Thus, there will be $c_i + 1$ cups of hot water and $c_i$ cups of cold water for some $c_i \geq 0$.
```

File: solutions/cf-1359C.mdx [hunk 20]
```diff
@@ -0,0 +1,114 @@
...
+- It can be proven, using induction, that the average barrel temperatures when $c$ is odd is a monotonically decreasing function. Thus, we can binary search on the maximum number of odd cups $c$ which gives a temperature of at least $t$ by checking the odd integers around it.
+
+The answer will be either be the barrel with $2$, $c$, or $c + 2$ cups.
```
Comment (maggie-j-liu) [inline solutions/cf-1359C.mdx:19]:
```suggestion
- It can be proven, using induction, that the average barrel temperatures when $c_i$ is odd is a monotonically decreasing function. Thus, we can binary search on the maximum number of odd cups $c_i$ which gives a temperature of at least $t$ by checking the odd integers around it.

The answer will be either be the barrel with $2$, $c_i$, or $c_i + 2$ cups.
```

File: solutions/cf-1359C.mdx (hunk 18 above)
Comment (maggie-j-liu) [reply solutions/cf-1359C.mdx:87]: i think this was me... dw, it is correct - github just makes tabs larger which makes the spacing seem off
image

**Review by maggie-j-liu — APPROVED** [review]

File: solutions/cf-1359C.mdx [hunk 21]
```diff
@@ -0,0 +1,112 @@
...
+		 */
+		double totalBarrel1 = (hot_water * (maximumCups + 1) + cold_water * maximumCups);
+		double totalBarrel2 = (hot_water * (maximumCups + 2) + cold_water * (maximumCups + 1));
+		double barrel1Cups = (double)(2 * maximumCups + 1.0);
+		double barrel2Cups = (double)(2 * maximumCups + 3.0);
+		double avgBarrel1 = totalBarrel1 / barrel1Cups;
+		double avgBarrel2 = totalBarrel2 / barrel2Cups;
+		double hot_cold_avg = (hot_water + cold_water) / 2.0;
+
+		double minAbsDifference = min(abs(desired_temp - avgBarrel1), min(abs(desired_temp - avgBarrel2), abs(desired_temp - hot_cold_avg)));
```
Comment (dongliu0426) [inline solutions/cf-1359C.mdx:85]: this line is > 80 characters

File: solutions/cf-1359C.mdx [hunk 22]
```diff
@@ -0,0 +1,112 @@
...
+[official editorial](https://codeforces.com/blog/entry/78116).**
+</Info>
+
+Consider the following observations:
+
+- Case 1: The number of cups of hot water poured is equal to the number of cups of cold water poured. Pouring an equal number of cups of hot and cold water is equivalent to pouring one cup of each. 
+- Case 2: Otherwise, there must be exactly one more cup of hot water poured than cold water poured. Thus, there will be $c_i + 1$ cups of hot water and $c_i$ cups of cold water for some $c_i \geq 0$.
+- It can be proven, using induction, that the average barrel temperatures when $c_i$ is odd is a monotonically decreasing function. Thus, we can binary search on the maximum number of odd cups $c_i$ which gives a temperature of at least $t$ by checking the odd integers around it.
+
+The answer will be either be the barrel with $2$, $c_i$, or $c_i + 2$ cups.
```
Comment (dongliu0426) [inline solutions/cf-1359C.mdx:19]: > will be either be

??

**Review by dongliu0426 — CHANGES_REQUESTED** [review]

File: solutions/cf-1359C.mdx (hunk 21 above)
Comment (jessechoe10) [reply solutions/cf-1359C.mdx:85]: ummm can you modify then...

File: solutions/cf-1359C.mdx (hunk 22 above)
Comment (jessechoe10) [reply solutions/cf-1359C.mdx:19]: oh shit mb

**Review by dongliu0426 — APPROVED** [review]
